analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionEmptinessCheckingCodeFix.cs

```diff
@@ -0,0 +1,100 @@
+            {
+                var binaryLeft = binary.Left;
+                var binaryRight = binary.Right;
+
+                if (language.ExpressionNumericConverter.TryGetConstantIntValue(binaryLeft, out var left))
+                {
+                    Simplify(root, binary, binaryRight, language.Syntax.ComparisonKind(binary).Mirror().Compare(left), context);
+                }
+                else if (language.ExpressionNumericConverter.TryGetConstantIntValue(binaryRight, out var right))
+                {
+                    Simplify(root, binary, binaryLeft, language.Syntax.ComparisonKind(binary).Compare(right), context);
+                }
+            }
+            return Task.CompletedTask;
+        }
+
+        public static void Simplify(SyntaxNode root, ExpressionSyntax expression, ExpressionSyntax countExpression, CountComparisonResult comparisonResult, CodeFixContext context) =>
+            context.RegisterCodeFix(
+                CodeAction.Create(Title, c => Replacement(root, expression, (InvocationExpressionSyntax)countExpression, comparisonResult, context)),
+                context.Diagnostics);
+
+        private static Task<Document> Replacement(SyntaxNode root, ExpressionSyntax expression, InvocationExpressionSyntax count, CountComparisonResult comparison, CodeFixContext context)
+        {
+            var any = NotAsExtenion(count)
+                ? AnyAsExtension(count)
+                : AnyFromExtension(count);
+
+            SyntaxNode replacement = comparison == CountComparisonResult.Empty
+                ? SyntaxFactory.PrefixUnaryExpression(SyntaxKind.LogicalNotExpression, any)
+                : any;
+
+            return Task.FromResult(context.Document.WithSyntaxRoot(root.ReplaceNode(expression, replacement).WithAdditionalAnnotations(Formatter.Annotation)));
+        }
+
+        private static InvocationExpressionSyntax AnyFromExtension(InvocationExpressionSyntax count)
+        {
+            var memberAccess = (MemberAccessExpressionSyntax)count.Expression;
+            var name = memberAccess.WithName(SyntaxFactory.IdentifierName(nameof(Enumerable.Any)));
+            return SyntaxFactory.InvocationExpression(name, count.ArgumentList);
+        }
+
+        private static InvocationExpressionSyntax AnyAsExtension(InvocationExpressionSyntax count)
+        {
+            var identifier = (IdentifierNameSyntax)count.ArgumentList.Arguments[0].Expression;
+            var name = SyntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, identifier, SyntaxFactory.IdentifierName(nameof(Enumerable.Any)));
+            var arguments = SyntaxFactory.ArgumentList(count.ArgumentList.Arguments.RemoveAt(0));
+            return SyntaxFactory.InvocationExpression(name, arguments);
+        }
+
+        private static bool NotAsExtenion(InvocationExpressionSyntax count)
```

Comment:
_Nitpick:_ Formatting
```suggestion
        private static bool NotAsExtenion(InvocationExpressionSyntax count) =>
```